internal/layer2/merge.go

```diff
@@ -0,0 +1,146 @@
+package layer2
+
+import (
+	"fmt"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	cloudConfigHeader = "#cloud-config"
+	jinjaHeader      = "## template: jinja"
+)
+
+// CloudConfigMerger handles the merging of cloud configs
+type CloudConfigMerger struct {
+}
+
+// NewCloudConfigMerger creates a new instance of CloudConfigMerger
+func NewCloudConfigMerger() *CloudConfigMerger {
+	return &CloudConfigMerger{}
+}
+
+// configHeaders represents the headers found in a cloud-config file
+type configHeaders struct {
+	hasJinja      bool
+	hasCloudConfig bool
+}
+
+// stripHeaders removes the template and cloud-config headers and returns the remaining content
+func stripHeaders(data string) (string, configHeaders) {
+	headers := configHeaders{}
+	lines := strings.Split(strings.TrimSpace(data), "\n")
+	startIndex := 0
+
+	for i, line := range lines {
+		trimmedLine := strings.TrimSpace(line)
+		switch trimmedLine {
```

Comment:
this assumes there will not be any other header !

Can we modify this to be more generic - is there a way to identify headers apart from string equals